### PR TITLE
LibGUI: Update progress of thumbnail generations on failure 

### PR DESCRIPTION
**Pull request by LucasChollet**
If you have some images which can't be decoded, the progress bar will hang in an unfinished state.
With this patch, even if we fail to generate a thumbnail, we update the progress to notify that we are done with the file.

WHF

### CODE REVIEW

File: Userland/Libraries/LibGUI/FileSystemModel.cpp [hunk 1]
```diff
@@ -699,12 +699,8 @@ bool FileSystemModel::fetch_thumbnail_for(Node const& node)
     auto const action = [path](auto&) {
         return render_thumbnail(path);
     };
-    auto const on_complete = [path, weak_this](auto thumbnail) -> ErrorOr<void> {
-        s_thumbnail_cache.with_locked([path, thumbnail](auto& cache) {
-            cache.thumbnail_cache.set(path, thumbnail);
-            cache.loading_thumbnails.remove(path);
-        });
 
+    auto const update_progress = [weak_this](bool with_sucess) {
```
Comment (mpurnell1) [inline Userland/Libraries/LibGUI/FileSystemModel.cpp:703]:
```suggestion
    auto const update_progress = [weak_this](bool with_success) {
```
Small typo fix

File: Userland/Libraries/LibGUI/FileSystemModel.cpp [hunk 2]
```diff
@@ -714,19 +710,36 @@ bool FileSystemModel::fetch_thumbnail_for(Node const& node)
                 strong_this->m_thumbnail_progress_total = 0;
             }
 
-            strong_this->did_update(UpdateFlag::DontInvalidateIndices);
+            if (with_sucess)
```
Comment (mpurnell1) [inline Userland/Libraries/LibGUI/FileSystemModel.cpp:713]:
```suggestion
            if (with_success)
```

**Review by mpurnell1 — CHANGES_REQUESTED** [review]
Nit: typo in spelling of "success"

File: Userland/Libraries/LibThreading/BackgroundAction.h [hunk 3]
```diff
@@ -98,8 +98,12 @@ class BackgroundAction final : public Core::Object
                     error = result.release_error();
 
                 m_promise->cancel(Error::from_errno(ECANCELED));
-                if (m_on_error)
-                    m_on_error(move(error));
+                if (m_on_error) {
+                    origin_event_loop->deferred_invoke([this, error = move(error)] {
+                        m_on_error(Error::copy(error));
```
Comment (linusg) [inline Userland/Libraries/LibThreading/BackgroundAction.h:103]: Why can we no longer `move()` the original error?

Comment (LucasChollet) [reply Userland/Libraries/LibThreading/BackgroundAction.h:103]: I don't know :shrug:, we discussed it a bit on Discord, I pinged you

**Pull request merged**